### DISCUSSION TITLE
Flexibility to handle 0 or 100% positive responses in a time slice

### DISCUSSION THREAD

**Discussion by Lewis-Barnett-NOAA** [sdmTMB/sdmTMB#304] [Ideas]
Maybe we already have this capability(?), but I was thinking about the [modifications made in VAST](https://github.com/search?q=repo%3AJames-Thorson-NOAA%2FVAST%20100%25%20encounter&type=code) that were necessary under that infrastructure to accommodate such data.  

We do run into the case with all positives in some years for Bering Sea Pacific cod.

**Comment by Lewis-Barnett-NOAA** [comment]
I suppose we'd have to add the complementary log-log link as an option.  Nice to see the description relative to the poisson-link in the new vignette.

**Comment by seananderson** [comment]
cloglog is already an option: https://github.com/pbs-assess/sdmTMB/blob/01ac913788e39f969b97a2ee9f1f749bddf9c010/src/sdmTMB.cpp#L30
although I'm not sure it's allowed as a valid link everywhere it should be or whether it works with all delta families yet. It should work with binomial() and was used in this paper https://cdnsciencepub.com/doi/full/10.1139/cjfas-2022-0279

Can you summarize what the approach was to 0 or 100% encounter rates? Mapping a parameter to a small or large value?

**Comment by Lewis-Barnett-NOAA** [comment]
Thanks Sean.  I'm looping in @James-Thorson-NOAA as he is more familiar with the details of how this was dealt with in VAST.  Previously, as I understand it, we would fit a delta model with the cloglog, but it also required a lot of if() clauses to detect which parameters had to be mapped differently.  I talked to Jim briefly about it today and it seems like we might be able to identify a cleaner way of doing it, perhaps leveraging the use of priors.
